plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/PinotQueryBuilder.java

```diff
@@ -167,9 +169,20 @@ private static Object convertValue(Type type, Object value)
         if (type instanceof VarbinaryType) {
             return Hex.encodeHexString(((Slice) value).getBytes());
         }
+        if (type instanceof TimestampType) {
+            return toMillis((Long) value);
+        }
         return value;
     }
 
+    private static Long toMillis(Long value)
+    {
+        if (value == null) {
+            return null;
+        }
+        return Timestamps.epochMicrosToMillisWithRounding(value);
```

Comment:
Can we add a type mapping test like we did for Druid ?

Reply:
Can you elaborate more?

Reply:
This value is from presto side, so it's a Long always.

Reply:
Actually we have a `TypeMapping` tests for each connector i.e we need something like this - https://github.com/trinodb/trino/blob/master/plugin/trino-druid/src/test/java/io/trino/plugin/druid/TestDruidTypeMapping.java

Reply:
There is no such typeMapping check here. I suggest to add it later.

Reply:
Can we add it as a preparatory commit ? It allows us to test the timestamp of various values.

Reply:
Pinot connector side internally uses this ALL_TYPES table for the testing with kafka.

Reply:
Added one more test on the time column type for dateTimeField in Pinot.